client/netdb.c

```diff
@@ -133,13 +134,14 @@ static struct host *random_host(int mask)
 
 static struct host *ipport2host(struct host *h, const char *ipport, int flags)
 {
-	unsigned ip[5];
+	unsigned ip[4];
+	unsigned short port;
 
-	if (sscanf(ipport, "%u.%u.%u.%u:%u", ip, ip + 1, ip + 2, ip + 3, ip + 4) != 5
-		|| (ip[0] | ip[1] | ip[2] | ip[3]) & ~0xff || ip[4] & ~0xffff) return 0;
+	if (sscanf(ipport, "%u.%u.%u.%u:%u", &ip[0], &ip[1], &ip[2], &ip[3], &port) != 5
+		|| !(ip[0] | ip[1] | ip[2] | ip[3])  || !port) return 0;
```

Comment:
You are doing it wrong, i think you should think more here

Reply:
for unsigned char value a
a & ~0xff == a & 0x00 == 0, i don't know why use this ?

Reply:
I think the logic was to decide whether ip is valid and port is a valid

Reply:
sorry it misunderstood unsigned as uint8_t